File: test_C/final/src/mouvement.c
#include <wiringPi.h>
#include <stdio.h>

#define GPIO_ENABLE1 23 //gpio 13
#define GPIO_IN1 29 //gpio 21
#define GPIO_IN2 25 // gpio 26
#define GPIO_ENABLE2 26 //gpio 12
#define GPIO_IN3 0 //gpio 17
#define GPIO_IN4 1 //gpio 18

void init() {
	
	pwmSetMode(PWM_MODE_MS);
	pwmSetClock(500); //50 hertz

	pinMode(GPIO_ENABLE1, OUTPUT);//Enable 1
	pinMode(GPIO_IN1, OUTPUT);//IN1
	pinMode(GPIO_IN2, OUTPUT);//IN2
	pinMode(GPIO_ENABLE2, OUTPUT);//Enable 2
	pinMode(GPIO_IN3, OUTPUT);//IN3
	pinMode(GPIO_IN4, OUTPUT);//IN4
}

void stopper() {
	
	digitalWrite(GPIO_IN1, LOW);
	digitalWrite(GPIO_IN2, LOW);
	digitalWrite(GPIO_IN3, LOW);
	digitalWrite(GPIO_IN4, LOW);
}

void avancer() {
	digitalWrite(GPIO_IN1, LOW);
	digitalWrite(GPIO_IN2, HIGH);
	digitalWrite(GPIO_IN3, LOW);
	digitalWrite(GPIO_IN4, HIGH);
}

void reculer() {
	digitalWrite(GPIO_IN1, HIGH);
	digitalWrite(GPIO_IN2, LOW);
	digitalWrite(GPIO_IN3, HIGH);
	digitalWrite(GPIO_IN4, LOW);
}

void tournerDroite() {
	digitalWrite(GPIO_IN1, HIGH);
	digitalWrite(GPIO_IN2, LOW);
	digitalWrite(GPIO_IN3, LOW);
	digitalWrite(GPIO_IN4, HIGH);
}

void tournerGauche() {
	digitalWrite(GPIO_IN1, LOW);
	digitalWrite(GPIO_IN2, HIGH);
	digitalWrite(GPIO_IN3, HIGH);
	digitalWrite(GPIO_IN4, LOW);
}

File: projet/moteurs/mouvement.py
import RPi.GPIO as GPIO

from config.constantes import GPIO_ENABLE1, GPIO_IN1, GPIO_IN2, GPIO_ENABLE2, GPIO_IN3, GPIO_IN4, PUISSANCE_MOTEUR_1, PUISSANCE_MOTEUR_2

pwm1 = None
pwm2 = None

def setup_moteurs():
    global pwm1, pwm2
    GPIO.setmode(GPIO.BCM)
    GPIO.setup(GPIO_ENABLE1, GPIO.OUT)
    GPIO.setup(GPIO_IN1, GPIO.OUT)
    GPIO.setup(GPIO_IN2, GPIO.OUT)
    GPIO.setup(GPIO_ENABLE2, GPIO.OUT)
    GPIO.setup(GPIO_IN3, GPIO.OUT)
    GPIO.setup(GPIO_IN4, GPIO.OUT)

    pwm1 = GPIO.PWM(GPIO_ENABLE1, 1000)
    pwm2 = GPIO.PWM(GPIO_ENABLE2, 1000)
    pwm1.start(0)
    pwm2.start(0)

def avancer():
    GPIO.output(GPIO_IN1, GPIO.HIGH)
    GPIO.output(GPIO_IN2, GPIO.LOW)
    GPIO.output(GPIO_IN3, GPIO.HIGH)
    GPIO.output(GPIO_IN4, GPIO.LOW)

def reculer():
    GPIO.output(GPIO_IN1, GPIO.LOW)
    GPIO.output(GPIO_IN2, GPIO.HIGH)
    GPIO.output(GPIO_IN3, GPIO.LOW)
    GPIO.output(GPIO_IN4, GPIO.HIGH)

def tourner_gauche():
    GPIO.output(GPIO_IN1, GPIO.HIGH)
    GPIO.output(GPIO_IN2, GPIO.LOW)
    GPIO.output(GPIO_IN3, GPIO.LOW)
    GPIO.output(GPIO_IN4, GPIO.HIGH)

def tourner_droite():
    GPIO.output(GPIO_IN1, GPIO.LOW)
    GPIO.output(GPIO_IN2, GPIO.HIGH)
    GPIO.output(GPIO_IN3, GPIO.HIGH)
    GPIO.output(GPIO_IN4, GPIO.LOW)

def arreter():
    GPIO.output(GPIO_IN1, GPIO.LOW)
    GPIO.output(GPIO_IN2, GPIO.LOW)
    GPIO.output(GPIO_IN3, GPIO.LOW)
    GPIO.output(GPIO_IN4, GPIO.LOW)

def regler_vitesse(moteur, vitesse):
    if moteur == 1 and pwm1 is not None:
        GPIO.output(GPIO_ENABLE1, GPIO.HIGH)
        pwm1.ChangeDutyCycle(vitesse)
    elif moteur == 2 and pwm2 is not None:
        GPIO.output(GPIO_ENABLE2, GPIO.HIGH)
        pwm2.ChangeDutyCycle(vitesse)